ui/perfherder/helpers.js

```diff
@@ -377,3 +389,394 @@ export const getGraphsLink = function getGraphsLink(
+        : thPerformanceBranches.filter(branch => branch !== alertRepository);
+    url += branches
+      .map(
+        branch =>
+          `&series=${branch},${alert.series_signature.signature_hash},0`,
+      )
+      .join('');
+  }
+
+  return url;
+};
+
+export const getSubtestsURL = (alert, alertSummary) => {
+  const endpoint = '#/comparesubtest';
+  const urlParameters = {
+    framework: alertSummary.framework,
+    originalProject: alertSummary.repository,
+    originalSignature: alert.series_signature.signature_hash,
+    newProject: alertSummary.repository,
+    newSignature: alert.series_signature.signature_hash,
+  };
+  if (alertSummary.prevResultSetMetadata) {
+    urlParameters.originalRevision =
+      alertSummary.prevResultSetMetadata.revision;
+  }
+  if (alertSummary.prevResultSetMetadata) {
+    urlParameters.newRevision = alertSummary.resultSetMetadata.revision;
+  }
+
+  return `${endpoint}${createQueryParams(urlParameters)}`;
+};
+
+const modifyAlert = (alert, modification) =>
+  update(getApiUrl(`/performance/alert/${alert.id}/`), modification);
+
+export const alertIsOfState = (alert, phAlertStatus) =>
+  alert.status === phAlertStatus.id;
+
+export const toggleStar = alert => {
+  const toggledStar = !alert.starred;
+  modifyAlert(alert, {
+    starred: toggledStar,
+  }).then(() => {
+    alert.starred = toggledStar;
+  });
+};
+
+let issueTrackers; // will cache on first AlertSummary call
+
+const _initializeAlerts = (alertSummary, optionCollectionMap) => {
+  // this function converts the representation returned by the perfherder
+  // api into a representation more suited for display in the UI
+
+  // just treat related (reassigned or downstream) alerts as one
+  // big block -- we'll display in the UI depending on their content
+  alertSummary.alerts = alertSummary.alerts
+    .concat(alertSummary.related_alerts)
+    .map(alertData => Alert(alertData, optionCollectionMap));
+};
+
+const AlertSummary = (alertSummaryData, optionCollectionMap) => {
+  if (issueTrackers === undefined) {
+    getData(getApiUrl(endpoints.issueTrackers)).then(
+      ({ data: issueTrackerList }) => {
+        issueTrackers = issueTrackerList;
+      },
+    );
+  }
+
+  const alertSummaryState = { ...alertSummaryData, issueTrackers };
+  _initializeAlerts(alertSummaryState, optionCollectionMap);
+
+  return alertSummaryState;
+};
+
+export const modifyAlertSummary = (alertSummary, modification) =>
+  update(
+    getApiUrl(`/performance/alertsummary/${alertSummary.id}/`),
+    modification,
+  );
+
+export const alertSummaryIsOfState = (alertSummary, phAlertSummaryStatus) =>
+  alertSummary.status === phAlertSummaryStatus.id;
+
+const updateAlertSummaryStatus = (alertSummary, newStatus) =>
+  update(getApiUrl(`/performance/alertsummary/${alertSummary.id}/`), {
+    status: newStatus.id,
+  }).then(() => {
+    alertSummary.status = newStatus.id;
+  });
+
+export const alertSummaryMarkAs = (alertSummary, phAlertSummaryStatus) => {
```

Comment:
This function appears to be unused, so can be removed?

Reply:
(Which would make `updateAlertSummaryStatus` unused too I think?)

Reply:
> 
> 
> This function appears to be unused, so can be removed?

Glad you asked! I didn't notice that. Turns out I've missed a couple of bounded alert summary methods from alertsctrl.html. Previously, these methods were generated at runtime (for ease of use), with the form "mark<newAlertSummaryState>()".
I had to do quite a bit of changes. This means the new commit needs a thorough code review as well.